odk2-src/build-app.rst

```diff
@@ -0,0 +1,778 @@
+
+.. csv-table:: Example Framework Survey Worksheet
+  :header: "branch_label", "url", "clause", "condition", "type", "values_list", "display.text", "display.hint"
+
+  "testForm",
+  , "''?' + opendatakit.getHashString('testForm')",,, "external_link",,"Open form",
+  ,,"exit section",
+
+The following changes will also need to be made to the :file:`framework.xlsx` **choices worksheet**
+
+.. csv-table:: Example Framework Choices Worksheet
+  :header: "choice_list_name", "data_value", "display.text"
+
+  "test_forms", "testForm", "testForm"
+
+The changes to the *choices* sheet add the *testForm* form as one of the choices that is shown in the :tc:`user_branch` prompt (a user-directed branching prompt type). The changes on the *survey* sheet add a branch label, :tc:`testForm`, that matches the :th:`data_value` from the *choices* sheet (this branch label will be jumped to if the user selects the :tc:`testForm` selection on the :tc:`user_branch` screen). The new branch label then renders an :tc:`external_link` prompt type that has the necessary arguments to open the *testForm*.
+
+
+.. _build-app-generate-formdef:
+
+Generating :file:`formDef.json`
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Once you have a saved your :file:`.xlsx` file, you can use the XLSX Converter to create a :file:`formDef.json`. Make sure your Application Designer is running (see :doc:`app-designer-launching`) and navigate to the :guilabel:`XLSX Converter` tab. Drag the :file:`.xlsx` form or select it with the :guilabel:`Choose File` button and use the :guilabel:`Save to File System` button to save the form definition file back to the file system.
+
+For the ongoing example, convert the :file:`app/config/assets/framework.xlsx` using the instructions above. Then repeat this process with :file:`app/config/tables/census/forms/census/census.xlsx`
+
+.. warning::
+
+  The :guilabel:`Save to File System` button uses the *form_id* and *table_id* within the :file:`.xlsx` file to identify where to write the :file:`formDef.json` file. If you have copied the :file:`.xlsx` file from some other location, and forgot to edit it, it may update back to that older location! If the *form_id* is equal to the *table_id*, two additional files are written that define the table's user data fields and that define the key-value properties for the table.
+
+Once you have made these changes and used XLSX Converter on the :file:`framework.xlsx` file to update the :file:`app/config/assets/framework/forms/framework/formDef.json` file, you should see your new form show up in the :guilabel:`Preview` tab of the Application Designer. Clicking on that should open your form.
+
+.. tip::
+
+  If you don't see your form in the :guilabel:`Preview`, try refreshing your browser.
+
```

Comment:
We can add another tip here that you can convert forms with `grunt xlsx-convert-all`.